### ISSUE TITLE
Remove White line on edge of HUD overlay.

### ISSUE THREAD

**Issue by DeputyLOL** [Mihara/RasterPropMonitor#372]
My hud has a black background. I wanted to get the diagonal shape of the vertical speed indicator so I had to cover it with a black shape in the HUD overlay. The problem I am having is that there is a line where the overlay becomes transparent to allow for the ladder and bars to show through.
(if that doesn't make sense then ill try to explain it better)

On the picture you can see the line just to the right of the speed indicator and below where the heading should be.

(also the -1 is the mechjeb space plane, still not working for me)

![2015-09-21_00001](https://cloud.githubusercontent.com/assets/12616126/9989369/be035b10-604f-11e5-9ab3-24bcc75ec3c9.jpg)

**Comment by MOARdV** [comment]
I don't see white lines on the hud textures I've got here, and I'm not applying white as far as I can tell to the texture.  I'll do some more digging, but I am not sure where that's coming from.

**Comment by DeputyLOL** [comment]
these are custom ones. It happens when you use a colour on the overlay next to a transparent section. Its much more noticeable and annoying with black. I had to use black to cover the corners of the verticle speed bar (far right) to get the 'triangular' shape of it. But there was an annoying white line in between the black background and the black on the overlay. So I covered everything in black and that did the trick. But now, I redesign the HUD and its come back.  
Its not a major problem.

BTW, my cockpit Is really coming to together, there is one feature that would allow me to finish the final screens. A round gauge that you can configure, that would allow me to build engine displays, a realistic radar altimeter and a roll indicator. Can you let me know if that is possible with the following features:
- The background and 'needles' should be from texture files so can customise them. 
- There can be multiple needles on one gauge showing different things.
- Two different types of needle: One is just a line that rotates to what you want.
                                               One that is a shape that covers the gauge background behind the needle.
- Able to enable/disable (like vertical bars)

If its too much to ask then I'm sorry and ill try work around it, but these would look really cool and open the gates for more designs in IVA.

**Comment by MOARdV** [comment]
Can you package that monitor / assets so I can try to reproduce it here?  I really don't have a path forward without being able to duplicate the issue locally.  As for the new feature - you're looking for something that draws on a regular MFD, but can simulate being a dial gauge, correct?  If so, yeah, that should be possible.  It might be a little while before I tackle it (I need to take a bit of a break from developing RPM, since work is getting busy).  I need to see what you mean with the second needle.  I understand the rotating needle, but I'm not sure what the other one should look like.

**Comment by DeputyLOL** [comment]
yes it is a dial gauge. Let me bring back this picture to explain the second needle (remember this???)

![ed diagram](https://cloud.githubusercontent.com/assets/12616126/10013318/69d7366a-6108-11e5-92c0-53c7033abd7c.png)

As you can see, the area behind the needle is filled in. That's may second needle.

**Comment by DeputyLOL** [comment]
also, if I can give you the overlay, would that help?
![adi_overlay3test](https://cloud.githubusercontent.com/assets/12616126/10013450/84956a3e-6109-11e5-8af9-493d5f4d6a1e.png)

**Comment by MOARdV** [comment]
Okay, I remember that image.  That's probably do-able.  I'll create an issue to track it.

With the white line, what I suspect is you're seeing a side-effect of texture sampling.  You're using white for your fill areas, then setting alpha to zero.  When the texture hardware on the GPU samples the data, it's doing a bilinear filter on the sample point, so it's collecting the white pixels with a blending alpha, creating a small line.  Using black (or a very dark gray) for your holes instead of white will likely eliminate the problem.

**Comment by DeputyLOL** [comment]
But everything that isn't black is transparent.
The only white in the texture are the lines that surround some of the bars. (the ones that ARE supposed to be there).
I use transparent because the bars show up behind the overlay and anything not transparent would block it. (as they should)

**Comment by MOARdV** [comment]
Every texel has an RGBA value.  In your case, if you open the texture with a program that doesn't support alpha channels (I have an ancient version of Paint Shop Pro), you'll see the holes as white.  So, even though you set alpha to 0, the RGB triplet on those texels is (1, 1, 1) (or (255, 255, 255) if you prefer).  What I suspect happens is when the texture hardware samples the point, it's getting some overlap between the black (0, 0, 0, 0) pixel and the white (1, 1, 1, 1) pixel, and it blends them into something like (0.5, 0.5, 0.5, 0.5).  One way to check this is to draw a black band around the edge of your transparency where you get the bleed-over (a 1 pixel black line should be enough).  If this solves the problem, replace the black line with something like a black line with low alpha (1/8 or lower).

I'll review the texture code again, but there aren't any places where I'm using white as a fill or boundary in this area of the code.

**Comment by DeputyLOL** [comment]
I tried that. when I first made the vertical it had to have a diagonal edge. But there was a white line, so I drew over it in black, but the white line was still there. My work around was to fill the entire texture black except for the transparent holes where the bars and ladder would show. It worked well until I needed to leave a gap for a bar with a transparent background.

I'm using paint.net btw.
